File: solutions/cpp/perfect-numbers/1/perfect_numbers.cpp
#include "perfect_numbers.h"
#include <stdexcept>
namespace perfect_numbers {
    long long aliquot_sum(long long value) {
        if (value == 1)
            return 0;

        long long aliquot_sum{ 1 };
        for (long long n = 2; n < value; ++n) {
            if (value % n == 0)
                aliquot_sum += n;
        }
        return aliquot_sum;
    }
    classification classify(long long value) {
        if (value < 1)
            throw std::domain_error("invalid value.");

        if (value < aliquot_sum(value))
            return classification::abundant;
        else if (value > aliquot_sum(value))
            return classification::deficient;
        else
            return classification::perfect;
    }

}  // namespace perfect_numbers

File: solutions/python/perfect-numbers/1/perfect_numbers.py
CLASS_PERFECT: str = "perfect"
CLASS_ABUNDANT: str = "abundant"
CLASS_DEFICIENT: str = "deficient"


def classify(number: int) -> str:
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")

    factors: list[int] = [x for x in range(1, number // 2 + 1) if number % x == 0]
    match sum(factors):
        case aliquot_sum if number < aliquot_sum: return CLASS_ABUNDANT
        case aliquot_sum if number > aliquot_sum: return CLASS_DEFICIENT
        case _:  return CLASS_PERFECT
